sql/moz-fx-data-shared-prod/marketing_campaign_names/asa_v1/udf.sql

```diff
@@ -0,0 +1,53 @@
+CREATE OR REPLACE FUNCTION marketing_campaign_names.asa_v1(campaign_name STRING)
+RETURNS ARRAY<STRUCT<key STRING, value STRING>> AS (
+  IF(
+    NOT (
+      REGEXP_CONTAINS(campaign_name, r'^asa_v1')
+      AND ARRAY_LENGTH(SPLIT(campaign_name, "_")) = 14
+    ),
+    NULL,
+    ARRAY(
+      SELECT AS STRUCT
```

Comment:
Would love alternate ideas on fast zipping of arrays. I did not have a chance to test or compare performance